prov/verbs/src/verbs_domain.c

```diff
@@ -89,12 +89,11 @@ fi_ibv_mr_reg(struct fid *fid, const void *buf, size_t len,
 	/* Enable local write access by default for FI_EP_RDM which hides local
 	 * registration requirements. This allows to avoid buffering or double
 	 * registration */
-	if (!(md->domain->info->caps & FI_LOCAL_MR) ||
-	    (md->domain->info->domain_attr->mr_mode & FI_MR_LOCAL))
+	if (!(md->domain->info->caps & FI_LOCAL_MR) &&
```

Comment:
It was an incorrect check.

The verbs/RDM doesn't require `FI_LOCAL_MR` and `FI_MR_LOCAL` MR bits set.
If we want to enable local write access by default for the FI_EP_RDM, we should check that neither `FI_LOCAL_MR` nor `FI_MR_LOCAL` bits are set.